### ISSUE TITLE
[Bug]: [ResourceGroup] Failed to get replica info after remove querynode pod failure chaos

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#22445]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.2.0-20230227-b01b467b
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka): pulsar   
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.2.3.dev1
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. deploy cluster with 4 querynodes
2. default rg has 2 available nodes (queryNodeID=5 and queryNodeID=3)
3. load collection `ResourceGroup_222` with 2 replicas into default rg (the default rg also has other collections)
query segment info and replica info:
```
[2023-02-27T11:31:18.353Z] [2023-02-27 11:31:17 - INFO - ci_test]: ResourceGroup_222 segment info: [segmentID: 439745809880911437

[2023-02-27T11:31:18.353Z] collectionID: 439745809880911427

[2023-02-27T11:31:18.353Z] partitionID: 439745809880911428

[2023-02-27T11:31:18.353Z] num_rows: 5035

[2023-02-27T11:31:18.353Z] state: Sealed

[2023-02-27T11:31:18.353Z] nodeIds: 5

[2023-02-27T11:31:18.353Z] nodeIds: 3

[2023-02-27T11:31:18.353Z] , segmentID: 439745809880911436

[2023-02-27T11:31:18.353Z] collectionID: 439745809880911427

[2023-02-27T11:31:18.353Z] partitionID: 439745809880911428

[2023-02-27T11:31:18.353Z] num_rows: 4965

[2023-02-27T11:31:18.353Z] state: Sealed

[2023-02-27T11:31:18.353Z] nodeIds: 5

[2023-02-27T11:31:18.353Z] nodeIds: 3

[2023-02-27T11:31:18.353Z] ] (test_chaos_resource_group.py:116)

[2023-02-27T11:31:18.353Z] [2023-02-27 11:31:17 - INFO - ci_test]: ResourceGroup_222 replica info: Replica groups:

[2023-02-27T11:31:18.353Z] - Group: <group_id:439745852861054983>, <group_nodes:(3,)>, <shards:[Shard: <channel_name:by-dev-rootcoord-dml_6_439745809880911427v0>, <shard_leader:3>, <shard_nodes:[3, 3, 3]>, Shard: <channel_name:by-dev-rootcoord-dml_7_439745809880911427v1>, <shard_leader:3>, <shard_nodes:[3, 3, 3]>]>

[2023-02-27T11:31:18.353Z] - Group: <group_id:439745852861054984>, <group_nodes:(5,)>, <shards:[Shard: <channel_name:by-dev-rootcoord-dml_6_439745809880911427v0>, <shard_leader:5>, <shard_nodes:[5, 5, 5]>, Shard: <channel_name:by-dev-rootcoord-dml_7_439745809880911427v1>, <shard_leader:5>, <shard_nodes:[5, 5, 5]>]> (test_chaos_resource_group.py:120)
```
5. pod failure `queryNodeID=5`  which belongs to the default rg
6. after 2m, remove chaos, pod becomes running, and new query node up `queryNodeID=17`
7. query node 17 online, but one replica of the `ResourceGroup_222` collection is not loaded into node 17, and get replicas info gets error:
```
[2023-02-27T11:42:28.661Z] [2023-02-27 11:42:28 - INFO - ci_test]: ResourceGroup_222 query segment info: [segmentID: 439745809880911437

[2023-02-27T11:42:28.661Z] collectionID: 439745809880911427

[2023-02-27T11:42:28.661Z] partitionID: 439745809880911428

[2023-02-27T11:42:28.661Z] num_rows: 5035

[2023-02-27T11:42:28.661Z] state: Sealed

[2023-02-27T11:42:28.661Z] nodeIds: 3

[2023-02-27T11:42:28.661Z] , segmentID: 439745809880911436

[2023-02-27T11:42:28.661Z] collectionID: 439745809880911427

[2023-02-27T11:42:28.661Z] partitionID: 439745809880911428

[2023-02-27T11:42:28.661Z] num_rows: 4965

[2023-02-27T11:42:28.661Z] state: Sealed

[2023-02-27T11:42:28.661Z] nodeIds: 3

[2023-02-27T11:42:28.661Z] ] (test_chaos_resource_group.py:180)

[2023-02-27T11:42:28.661Z] pymilvus.exceptions.MilvusException: <MilvusException: (code=15, message=failed to get replica info, err=failed to get shard leader for shard collectionID:439745809880911427 channelName:"by-dev-rootcoord-dml_6_439745809880911427v0" seek_position:<channel_name:"by-dev-rootcoord-dml_6_439745809880911427v0" msgID:"\010\010\020I\030\000 \000" msgGroup:"by-dev-dataNode-16-by-dev-rootcoord-dml_6_439745809880911427v0" timestamp:439745898898259969 > flushedSegmentIds:439745809880911436 , the collection not loaded or leader is offline[NodeNotFound(0)])>
```

### Expected Behavior

query node 17 online and loaded replica `ResourceGroup_222` automatically
get replicas and return 2 replicas info

### Steps To Reproduce

```markdown
Jenkins pipeline:
chaos test -> chaos-test-resource-group -> pod-failure querynode
```

### Milvus Log

Jenkins job:  https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test-resource-group/detail/chaos-test-resource-group/65/pipeline/313

server pods:
```
19:46:32  querynode-pod-failure-65-milvus-datacoord-75b4c885bf-6vqcx        1/1     Running       0               22m     10.102.10.69    devops-node20   <none>           <none>
19:46:32  querynode-pod-failure-65-milvus-datacoord-75b4c885bf-zg576        1/1     Running       0               22m     10.102.7.145    devops-node11   <none>           <none>
19:46:32  querynode-pod-failure-65-milvus-datanode-676c789f97-b9p2m         1/1     Running       1 (18m ago)     22m     10.102.9.53     devops-node13   <none>           <none>
19:46:32  querynode-pod-failure-65-milvus-datanode-676c789f97-ftpnz         1/1     Running       0               22m     10.102.7.141    devops-node11   <none>           <none>
19:46:32  querynode-pod-failure-65-milvus-indexcoord-747df7959c-mg4hs       1/1     Running       0               22m     10.102.7.137    devops-node11   <none>           <none>
19:46:32  querynode-pod-failure-65-milvus-indexcoord-747df7959c-wxpsp       1/1     Running       0               22m     10.102.5.19     devops-node21   <none>           <none>
19:46:32  querynode-pod-failure-65-milvus-indexnode-7b66b9955d-mbnbw        1/1     Running       0               22m     10.102.5.20     devops-node21   <none>           <none>
19:46:32  querynode-pod-failure-65-milvus-proxy-6f747ffdbf-rw8pp            1/1     Running       1 (18m ago)     22m     10.102.5.24     devops-node21   <none>           <none>
19:46:32  querynode-pod-failure-65-milvus-querycoord-958dd464b-cw2hn        1/1     Running       0               22m     10.102.7.136    devops-node11   <none>           <none>
19:46:32  querynode-pod-failure-65-milvus-querycoord-958dd464b-zkbv6        1/1     Running       1 (18m ago)     22m     10.102.5.18     devops-node21   <none>           <none>
19:46:32  querynode-pod-failure-65-milvus-querynode-7787d8b45b-cc2nx        1/1     Running       0               22m     10.102.10.70    devops-node20   <none>           <none>
19:46:32  querynode-pod-failure-65-milvus-querynode-7787d8b45b-nbqnd        1/1     Running       0               22m     10.102.5.21     devops-node21   <none>           <none>
19:46:32  querynode-pod-failure-65-milvus-querynode-7787d8b45b-vnftx        1/1     Running       0               22m     10.102.6.24     devops-node10   <none>           <none>
19:46:32  querynode-pod-failure-65-milvus-querynode-7787d8b45b-xb7xg        1/1     Running       2 (12m ago)     22m     10.102.7.142    devops-node11   <none>           <none>
19:46:32  querynode-pod-failure-65-milvus-rootcoord-649856fcf4-bl5j4        1/1     Running       0               22m     10.102.7.143    devops-node11   <none>           <none>
19:46:32  querynode-pod-failure-65-milvus-rootcoord-649856fcf4-wrmzx        1/1     Running       0               22m     10.102.5.28     devops-node21   <none>           <none> 
```

log files:
[artifacts-querynode-pod-failure-65-server-logs.tar.gz](https://github.com/milvus-io/milvus/files/10839802/artifacts-querynode-pod-failure-65-server-logs.tar.gz)

[artifacts-querynode-pod-failure-65-pytest-logs.tar.gz](https://github.com/milvus-io/milvus/files/10839805/artifacts-querynode-pod-failure-65-pytest-logs.tar.gz)

**Comment by weiliu1031** [comment]
please verify this

**Comment by ThreadDao** [comment]
Fixed `2.2.0-20230301-820ec0fc`
